integration-tests/src/test_loop/builder.rs

```diff
@@ -119,7 +119,7 @@ impl TestLoopBuilder {
         }
         self.setup_network(&datas, &network_adapters, &epoch_manager_adapters);
 
-        let env = TestLoopEnv { test_loop: self.test_loop, datas };
+        let env = TestLoopEnv { test_loop: self.test_loop, datas, tempdir };
```

Comment:
This was not fun to debug. Without storing the tempdir, it was getting removed from the filesystem at the end of the scope here. A short while later the contract call would fail because the contract cache attempts to write to the subdirectory of the tempdir.